packages/react/src/utilities/selection/SelectionZone.tsx

```diff
@@ -765,7 +813,19 @@ export class SelectionZone extends React.Component<ISelectionZoneProps, ISelecti
   }
 
   private _getItemIndex(itemRoot: HTMLElement): number {
-    return Number(itemRoot.getAttribute(SELECTION_INDEX_ATTRIBUTE_NAME));
+    const indexValue = itemRoot.getAttribute(SELECTION_INDEX_ATTRIBUTE_NAME);
+
+    return Number(indexValue);
+  }
+
+  private _getItemSpan(itemRoot: HTMLElement): number | undefined {
+    const spanValue = itemRoot.getAttribute(SELECTION_SPAN_ATTRIBUTE_NAME);
+
+    if (typeof spanValue === 'string') {
+      return Number(spanValue);
```

Comment:
It's possible that parsing `spanValue` gives `NaN`.
```suggestion
      const returnValue = Number(spanValue);
      return isNaN(returnValue) ? undefined : returnValue;
```